src/main/scala/firrtl/transforms/ConstantPropagation.scala

```diff
@@ -207,7 +207,10 @@ class ConstantPropagation extends Transform with DependencyAPIMigration with Res
   }
 
   object FoldAND extends FoldCommutativeOp {
-    def fold(c1: Literal, c2: Literal) = UIntLiteral(c1.value & c2.value, c1.width max c2.width)
+    def fold(c1: Literal, c2: Literal) = {
+      val width = (c1.width max c2.width).asInstanceOf[IntWidth]
+      UIntLiteral((c1.value & c2.value) & ((BigInt(1) << width.width.toInt) - 1), width)
+    }
```

Comment:
This logic looks good, since it's written 3 times, I think we should probably factor this logic into a common utility function. Arguably it makes sense to override the apply method with this, but as an alternative:
```scala
// In ir/IR.scala
object UIntLiteral {
  /** Utility to construct UIntLiterals masked by the width
    *
    * This supports truncating negative values as well as values that are too wide for the width
    */
 def  masked(value: BigInt, width: IntWidth): UIntLiteral = {
    val mask = (BigInt(1) << width.width.toInt) - 1
   UIntLiteral(value & mask, width)
  }
}
```